pkgs/stdenv/cross/default.nix

```diff
@@ -48,7 +48,9 @@ in lib.init bootStages ++ [
       # Prior overrides are surely not valid as packages built with this run on
       # a different platform, and so are disabled.
       overrides = _: _: {};
-      extraBuildInputs = [ ]; # Old ones run on wrong platform
+      extraBuildInputs = [ ] # Old ones run on wrong platform
+         ++ lib.optionals hostPlatform.isDarwin [ buildPackages.targetPackages.darwin.apple_sdk.frameworks.CoreFoundation ]
```

Comment:
This is something I have found no good solution for, I'm hoping I missed something obvious. This step defines the stdenv that produces `targetPackages`, so including a dependency package from `targetPackages` is trivially infinite recursion. In the Darwin stdenv it's easy to build things on a different stage and use them in a later stage, since all platforms are the same. However when cross compiling the adjacent stages are quite rigid. The only way this isn't infinite recursion is by overriding `extraBuildInputs = [ ]` for `stdenvNoCC`, and using that to build `CoreFoundation`.